### PR TITLE
Devise

### PR DESCRIPTION
**Pull request by Draxeytina**
- [x] Remove current_user method in ApplicationController, and use devise one.

- [x] Install devise:

- Can register a new user.
- User logs in with a combination of email and password.
- Hashed password should be stored in the database.
- Ask for confirmation of email.
- Can reset password.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,5 +1,17 @@
 class ApplicationController < ActionController::Base
-  def current_user
```
Comment (mateo951) [inline app/controllers/application_controller.rb:2]: The current_user method in ApplicationController has been removed ✔

File: app/models/user.rb [hunk 2]
```diff
@@ -1,12 +1,17 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
```
Comment (mateo951) [inline app/models/user.rb:4]: The application allows for registering a new user 🚀

File: db/migrate/20220928220555_add_devise_to_users.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class AddDeviseToUsers < ActiveRecord::Migration[7.0]
+  def self.up
+    change_table :users do |t|
+      ## Database authenticatable
+      t.string :email,              null: false, default: ""
+      t.string :encrypted_password, null: false, default: ""
```
Comment (mateo951) [inline db/migrate/20220928220555_add_devise_to_users.rb:8]: A hashed password is stored in the database 💪😎

File: app/models/user.rb [hunk 4]
```diff
@@ -1,12 +1,17 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
Comment (mateo951) [inline app/models/user.rb:3]: Please be sure that the application asks for confirmation of email as it is part of the project requirements 📝 

Your model should look something like:
```ruby
devise :database_authenticatable, :registerable,
				:recoverable, :rememberable, :validatable, :confirmable
```

**Review by mateo951 — CHANGES_REQUESTED** [review]
### Hi @Draxeytina, 

**🔥Highlights🔥**
- The user can log in with a combination of email and password ✔
- The user can reset the password in the application ✔

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

File: app/models/user.rb (hunk 4 above)
Comment (Draxeytina) [reply app/models/user.rb:3]: Thank you @mateo951. I added the feature.

**Review by EricMbouwe — APPROVED** [review]
Hi @Draxeytina, 

Your project is complete! ⚡🔥  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🎉🔥
- Add requested changes ✔
- Linters ✔

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**